### PR TITLE
Add documentation for invalid-character-carriage-return

### PR DESCRIPTION
**Pull request by ollie-iterators**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Add documentation for invalid-character-carriage-return. NOTE: I have no idea why syntax-error is not being disabled.

Refs #5953 

Closes #XXXX

### CODE REVIEW

File: doc/data/messages/i/invalid-character-carriage-return/bad.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# pylint: disable=trailing-whitespace, missing-module-docstring, trailing-newlines, missing-final-newline, syntax-error
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/invalid-character-carriage-return/bad.py:1]: You don't need that, all message except the one we document and fatal one are disabled :)

Comment (ollie-iterators) [reply doc/data/messages/i/invalid-character-carriage-return/bad.py]: Changed it

**Comment by Pierre-Sassoulas** [comment]
@ollie-iterators there's a genuine fail here, it's not only ``self-import``

File: doc/data/messages/i/invalid-character-carriage-return/bad.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# pylint: disable = syntax-error
```
Comment (mbyrnepr2) [inline doc/data/messages/i/invalid-character-carriage-return/bad.py:1]:
```suggestion
```

**Comment by mbyrnepr2** [comment]
There's an [example](https://github.com/PyCQA/pylint/blob/aaa3770d325223ee5c13a637f09ff16e5bbc5851/tests/functional/b/bad_char/bad_char_carriage_return.py) in the functional tests which could be re-used.

**Comment by ollie-iterators** [comment]
> There's an [example](https://github.com/PyCQA/pylint/blob/aaa3770d325223ee5c13a637f09ff16e5bbc5851/tests/functional/b/bad_char/bad_char_carriage_return.py) in the functional tests which could be re-used.
> 
> I think the `disable`s can be omitted from the good/bad examples because these other messages are ignored in the examples.

I have tried to copy the example, but it does not seem to get copied.

**Pull request closed without merging**